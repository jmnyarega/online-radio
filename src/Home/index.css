.main-content {
  padding: 2% 2em 0 2em;
}

.hero-intro {
  margin-bottom: 8rem;
}

.hero__title {
  text-align: center;
}

.hero-text {
  text-align: center;
  color: var(--lightAccent);
}

.hero-text__intro {
  line-height: 1.6;
}

.hero-text__title {
  line-height: 1.2;
}

.hero-btn {
  float: right;
}

.mostlistened-container {
  padding: 0;
}

.mostlistened-container-list {
  font-size: 0.7rem;
  list-style: none;
  height: 50px;
  background-color: var(--lightAccent);
  filter: drop-shadow(0px 4px 4px var(--darkAccent));
  text-decoration: none;

  margin-top: 1em;
  padding: 0 1em;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.mostlistened-container-list:hover {
  opacity: 0.8;
}

.mostlistened__play {
  color: var(--darkAccent);
}

.mostlistened__item {
  justify-content: space-between;
}

.mostlistened__radio-name {
  color: var(--darkAccent);
  font-size: 10px;
  text-transform: capitalize;
  width: 5rem;
}

.mostlistened__rate {
  color: var(--gray);
}

.mostlistened__country {
  color: var(--gray);
  width: 5rem;
}

.mostlistened-row__line {
  position: relative;
  min-width: 300px;
}

.mostlistened__line {
  color: var(--lightAccent);
  font-weight: 900;
  text-decoration: none;
  cursor: pointer;
  position: absolute;
  bottom: 35%;
  right: 25%;
  height: 5px;
  padding-left: 0.5rem;
}

@media (min-width: 900px) {
  .main-content {
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
  }
  .hero-toplist {
    width: 45%;
  }
  .hero-intro {
    width: 45%;
    margin-bottom: 0;
  }
  .mostlistened-container-list {
    font-size: 1rem;
  }
}
