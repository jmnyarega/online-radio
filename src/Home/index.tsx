import React from "react";
import { Link } from "react-router-dom";
import Disc from "../Common/disc";
import "./index.css";

type Iprops = {
  mostlistened: Array<number>;
};

export const Home = ({ mostlistened }: Iprops): any => (
  <div className="container">
    <div className="main main__background">
      <div className="main-content">
        <div className="col hero hero-intro">
          <h3 className="title title__large hero-text hero-text__title">
            Listen to a variety of radio stations all over the world
          </h3>
          <p className="hero-text hero-text__intro">
            The easiest way to understand the culture of a country is to listen
            to their most popular radio station. There is a lot of information
            that is being shared on radio; news, music, way of living, politics,
            etc. Radio.radio is allowing you to experience and understand
            different cultures
          </p>
          <Link to="/discover" className="btn btn-hero hero-btn">
            Discover
          </Link>
        </div>
        <div className="col hero-toplist">
          <div className="row dv-center">
            <Disc className="col" />
            <h4 className="title title__small col">Most listened Stations</h4>
          </div>
          <div className="mostlistened">
            {mostlistened.map((x) => (
              <iframe
                key={x}
                title={`home radio ${x}`}
                src={`https://tunein.com/embed/player/s2972${x}/`}
                scrolling="no"
                style={{ width: "100%", height: "100px" }}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default Home;
