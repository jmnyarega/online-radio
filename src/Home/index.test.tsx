import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter as Router } from "react-router-dom";
import Home from "./index";

const mostlistened = [0, 2, 6, 8];

describe("renders components without crashing", () => {
  it("Home renders without crashing", () => {
    expect(shallow(<Home mostlistened={mostlistened} />)).toHaveLength(1);
  });
});

describe("Test home urls", () => {
  const wrapper = mount(
    <Router>
      <Home mostlistened={mostlistened} />{" "}
    </Router>
  );
  it("test most listened urls", () => {
    expect(wrapper.find("iframe")).toHaveLength(mostlistened.length);
  });
});

describe("snapshots", () => {
  const tree = toJson(shallow(<Home mostlistened={mostlistened} />));
  expect(tree).toMatchSnapshot();
});
