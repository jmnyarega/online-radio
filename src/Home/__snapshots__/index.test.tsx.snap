// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="container"
>
  <div
    className="main main__background"
  >
    <div
      className="main-content"
    >
      <div
        className="col hero hero-intro"
      >
        <h3
          className="title title__large hero-text hero-text__title"
        >
          Listen to a variety of radio stations all over the world
        </h3>
        <p
          className="hero-text hero-text__intro"
        >
          The easiest way to understand the culture of a country is to listen to their most popular radio station. There is a lot of information that is being shared on radio; news, music, way of living, politics, etc. Radio.radio is allowing you to experience and understand different cultures
        </p>
        <Link
          className="btn btn-hero hero-btn"
          to="/discover"
        >
          Discover
        </Link>
      </div>
      <div
        className="col hero-toplist"
      >
        <div
          className="row dv-center"
        >
          <Disc
            className="col"
          />
          <h4
            className="title title__small col"
          >
            Most listened Stations
          </h4>
        </div>
        <div
          className="mostlistened"
        >
          <iframe
            key="0"
            scrolling="no"
            src="https://tunein.com/embed/player/s29720/"
            style={
              Object {
                "height": "100px",
                "width": "100%",
              }
            }
            title="home radio 0"
          />
          <iframe
            key="2"
            scrolling="no"
            src="https://tunein.com/embed/player/s29722/"
            style={
              Object {
                "height": "100px",
                "width": "100%",
              }
            }
            title="home radio 2"
          />
          <iframe
            key="6"
            scrolling="no"
            src="https://tunein.com/embed/player/s29726/"
            style={
              Object {
                "height": "100px",
                "width": "100%",
              }
            }
            title="home radio 6"
          />
          <iframe
            key="8"
            scrolling="no"
            src="https://tunein.com/embed/player/s29728/"
            style={
              Object {
                "height": "100px",
                "width": "100%",
              }
            }
            title="home radio 8"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
